test/Sema/enum_raw_representable_generic.swift

```diff
@@ -0,0 +1,23 @@
+// RUN: %target-run-simple-swift
+
+// REQUIRES: executable-test
+
+enum Foo<T : ExpressibleByIntegerLiteral> : T where T:Equatable {
+  case a = 1
+  case b = 10
+  case c = 42
+}
+
+let _ : Int8  = Foo.a.rawValue
+let _ : Int16 = Foo.a.rawValue
+let _ : Int32 = Foo.a.rawValue
+let _ : Int64 = Foo.a.rawValue
+let _ : UInt8  = Foo.a.rawValue
+let _ : UInt16 = Foo.a.rawValue
+let _ : UInt32 = Foo.a.rawValue
+let _ : UInt64 = Foo.a.rawValue
+
+guard case .c = Foo(rawValue: UInt8(42))!  else { fatalError() }
+guard case .c = Foo(rawValue: UInt16(42))! else { fatalError() }
+guard case .c = Foo(rawValue: UInt32(42))! else { fatalError() }
+guard case .c = Foo(rawValue: UInt64(42))! else { fatalError() }
```

Comment:
Nice. It looks like the crash occurs because its trying to type check a function with no body, which is not allowed.

Can we push the check for 'all elements have the same raw type' up a level, so that we don't attempt to derive the rawRepresentable conformance at all in that case?

Similar to how we don't derive an Equatable conformance for enums that have payload cases (we could, eventually, but the point is we have control over when to derive the conformance).